temperature/router.py

```diff
@@ -0,0 +1,35 @@
+from fastapi import APIRouter, Depends, Query, HTTPException
+from sqlalchemy.orm import Session
+
+from dependencies import get_db
+from temperature import crud, schemas
+
+
+router = APIRouter()
+
+
+@router.get(
+    "/temperature/",
+    response_model=list[schemas.Temperature]
+)
+async def get_temperature_list(
+        city_id: int = Query(None, description="City id"),
+        db: Session = Depends(get_db)
+):
+
+    if city_id:
+        temperature = await crud.get_temperatures_by_city_id(
+            db=db, city_id=city_id
+        )
+        if not temperature:
+            raise HTTPException(status_code=400, detail="Temperature does not exist")
```

Comment:
The HTTP status code for a non-existent temperature is set to 400. Consider using a 404 status code to indicate that the resource was not found, which is more semantically correct.